### PR TITLE
Replace dead link with archive.org link

### PR DESCRIPTION
**Pull request by louwers**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The following services are available via HTTP API, C++ library interface and Nod
 
 To quickly try OSRM use our [demo server](http://map.project-osrm.org) which comes with both the backend and a frontend on top.
 
-For a quick introduction about how the road network is represented in OpenStreetMap and how to map specific road network features have a look at [this guide about mapping for navigation](https://www.mapbox.com/mapping/mapping-for-navigation/).
+For a quick introduction about how the road network is represented in OpenStreetMap and how to map specific road network features have a look at [this guide about mapping for navigation](https://web.archive.org/web/20221206013651/https://labs.mapbox.com/mapping/mapping-for-navigation/).
```
Comment (SiarheiFedartsou) [inline README.md:18]: May be worth putting link to some "official" OSM wiki page here? E.g. https://wiki.openstreetmap.org/wiki/Routing ?

Comment (DennisOSRM) [reply README.md:18]: +1

Comment (louwers) [reply README.md]: I think the old Mapbox guide is pretty useful as well still. I added a link to the wiki as well. What do you think?

Comment (SiarheiFedartsou) [reply README.md]: LGTM :+1:

**Review by SiarheiFedartsou — APPROVED** [review]

**Pull request merged**